From 42dad741607d8889f93b2a12725af362c09732a2 Mon Sep 17 00:00:00 2001
From: Simon Ser <contact@emersion.fr>
Date: Sun, 10 Jan 2021 12:24:04 +0100
Subject: [PATCH] Match #define when extracting DRM modifiers

Newer drm_fourcc.h headers have comments which refer to
DRM_FORMAT_MOD_*. These were matched by the regex.

Fix the regex by requiring a "#define" prefix. I checked that only the
duplicate values were removed using diff.
---
 fourcc.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/fourcc.py b/fourcc.py
index 23f7be6..ebee1d9 100755
--- a/fourcc.py
+++ b/fourcc.py
@@ -22,8 +22,8 @@ def afbc_print(f, l):
 
 info = {
 	'fmt': r'^#define (\w+)\s*(?:\\$\s*)?fourcc_code',
-	'basic_pre': r'\bI915_FORMAT_MOD_\w+\b',
-	'basic_post': r'\b(DRM_FORMAT_MOD_(?:INVALID|LINEAR|SAMSUNG|QCOM|VIVANTE|NVIDIA|BROADCOM|ALLWINNER)\w*)\s',
+	'basic_pre': r'^#define (I915_FORMAT_MOD_\w+)\b',
+	'basic_post': r'^#define (DRM_FORMAT_MOD_(?:INVALID|LINEAR|SAMSUNG|QCOM|VIVANTE|NVIDIA|BROADCOM|ALLWINNER)\w*)\s',
 	'afbc_block': r'\bAFBC_FORMAT_MOD_BLOCK_SIZE(?:_\d+x\d+)+\b',
 	'afbc_bitmask': r'\bAFBC_FORMAT_MOD_[A-Z]+\b',
 }
